File: EngineLib/src/Console.cpp
//
// Created by stumi on 11/11/21.
//
#include "Console.h"

#include <iostream>

namespace Engine
{
    void Console::clearBuffer() { std::cout << "clearBuffer" << std::endl; }

    void Console::flush() { std::cout << "flush" << std::endl; }

    void Console::putString(std::string str) { std::cout << str << std::endl; }

}  // namespace Engine

File: CSharpGameLib/CSharpGameLib/src/Console.cs
using System.Runtime.CompilerServices;

namespace Engine
{
    public class Console
    {
        [MethodImpl(MethodImplOptions.InternalCall)]
        public static extern void clearBuffer();
        

        [MethodImpl(MethodImplOptions.InternalCall)]
        public static extern void putString(string str);

        [MethodImpl(MethodImplOptions.InternalCall)]
        public static extern void flush();
    }
}